components/mpas-ocean/src/analysis_members/mpas_ocn_moc_streamfunction.F

```diff
@@ -508,10 +508,15 @@ subroutine ocn_compute_moc_streamfunction(domain, timeLevel, err)!{{{
                      * 0.5_RKIND*(layerThickness(k,c1) + layerThickness(k,c2))
                end do
             end do
-            do k = 2, nVertLevels
+            !Integrate transport from bottom to top, which switches the sign of sumTransport
+            !there is also an implicit moc(nVertLevels+1,iTransect) = 0
+            k = nVertLevels
+            mocStreamValLatAndDepthRegionLocal(1, k, iTransect) = &
+               - sumTransport(k, iTransect)
+            do k = nVertLevels-1, 1, -1
                mocStreamValLatAndDepthRegionLocal(1, k, iTransect) = &
-                  mocStreamValLatAndDepthRegionLocal(1, k - 1, iTransect) &
-                  + sumTransport(k - 1, iTransect)
+                  mocStreamValLatAndDepthRegionLocal(1, k + 1, iTransect) &
+                  - sumTransport(k + 1, iTransect)
             end do
          end do
```

Comment:
@vanroekel, I'm looking at this carefully to implement it in MPAS-Analysis.

It seems to me that 
```
mocStreamValLatAndDepthRegionLocal(1, nVertLevels, iTransect)
```
should be zero, not 
```
-sumTransport(nVertLevels, iTransect)
```

As written, 
```
mocStreamValLatAndDepthRegionLocal(1, nVertLevels-1, iTransect) = -2*sumTransport(nVertLevels, iTransect)
```
which doesn't seem right to me.

Reply:
@xylar I wrote it this way assuming there is an implicit `mocStreamValLatAndDepthRegionLocal(1,nVertLevels+1,tran) = 0`.  But looking again I think this means the second line is not right. we should have 
```
mocStreamValLatAndDepthRegionLocal(1, k + 1, iTransect) &
                  - sumTransport(k, iTransect)
 ```
instead of k+1.  Does this seem reasonable to you?  if so I can fix this in the PR.

Reply:
Yes, looking at it again, I think you are correct.

Reply:
In MPAS-Analysis, we are explicit about having `nVertLevels+1` entries in the MOC field, with a zeroes currently at the top and soon to be at the bottom.

I think the change you suggest would make this code consistent with POP, which you linked to above in the PR.

Reply:
Would you suggest I make the moc variable here nVertLevelsP1 for clarity?  I would be happy to make the change.

Reply:
I don't think that's necessary.  All the output variables are using `nVertLevels`:
https://github.com/E3SM-Project/E3SM/blob/e3f2c1524fb1a3a5cab0c283e2572064dea292e6/components/mpas-ocean/src/analysis_members/Registry_moc_streamfunction.xml#L54-L79
So it makes sense to stick with that same approach.  Conceptually, I think the extra row is a good idea because it make clear that the MOC values live between layers rather than at the centers of layers but I don't think it makes sense to change the size of the output variables, and thus I don't think we want to change the size of the temporary variables either.